carp.common/src/commonMain/kotlin/dk/cachet/carp/common/infrastructure/services/ApplicationServiceRequest.kt

```diff
@@ -0,0 +1,13 @@
+package dk.cachet.carp.common.infrastructure.services
+
+import kotlinx.serialization.KSerializer
+
+
+/**
+ * A request for [TService] stored in memory, which can be invoked using [invokeOn].
+ */
+interface ApplicationServiceRequest<TService, out TReturn>
+{
+    fun getResponseSerializer(): KSerializer<out TReturn>
+    suspend fun invokeOn( service: TService ): TReturn
```

Comment:
Thank you. This seems more right and is easier to reason about than the previous one via constructor delegation 👍